src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -127,14 +130,15 @@ private static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Sp
                     // Therefore, (destLength / 3) * 4 will always be less than 2147483641
                     Debug.Assert(destLength < (int.MaxValue / 4 * 3));
                     maxSrcLength = (destLength / 3) * 4;
+                    srcLength &= ~0x3; // Round down to multiple of 4, this only affect Base64UrlDecoder path
```

Comment:
> This code will be executed in both, right?

Yes

> And the comment is just saying it'll be a nop for Base64 because it'll always be a multiple of 4?

No, not NOP, for Base64 `srcLength` will be already rounded down to multiple of 4 at row 5 above. Just meant no value change

> Can/should we only even have this code if typeof(TBase64Decoder) == typeof(Base64UrlDecoder)?

Not sure having a condition worth when there is no value/functional change for Base64, if you prefer that then I can change